app/bundles/MessengerBundle/MessageHandler/PageHitNotificationHandler.php

```diff
@@ -0,0 +1,103 @@
+        } catch (InvalidPayloadException $payloadException) {
+            $this->logger->error('Invalid payload #'.$message->getHitId().' '.$payloadException->getMessage());
+
+            throw $payloadException;
+        } catch (\Exception $exception) {
+            $this->logger->error($exception->getMessage(), (array) $exception);
+            throw new MauticMessengerException($exception->getMessage(), 400, $exception);
+        }
+        $this->logger->info('processed page hit #'.$message->getHitId());
+    }
+
+    /** @return iterable<string, mixed> */
+    public static function getHandledMessages(): iterable
+    {
+        yield PageHitNotification::class => [];
+    }
+
+    /**
+     * @return array<string, mixed>
+     *
+     * @throws InvalidPayloadException
+     */
+    private function parseMessage(PageHitNotification $message): array
+    {
+        $hit = $message->getHitId() > 0 ? $this->hitRepository->find($message->getHitId()) : null;
+
+        $pageObject = null;
+        if (null !== $message->getPageId()) {
+            try {
+                $pageObject = $message->isRedirect()
+                    ? $this->redirectRepository->find($message->getPageId())
+                    : $this->pageRepository->find($message->getPageId());
+            } catch (\Exception $exception) {
+                $this->logger->error(
+                    sprintf('Invalid page/redirect, exception. #%s', $message->getPageId()), ['message' => $message]);
+                throw $exception;
+            }
+
+            if (null === $pageObject) {
+                $this->logger->error(
+                    sprintf('Invalid page/redirect, id not found. #%s', $message->getPageId())
+                );
+                throw new InvalidPayloadException('Missing required information', ['message' => $message]);
+            }
+        }
+
+        if (!$hit instanceof Hit && $message->getHitId() > 0) {
+            $this->logger->warning('Invalid hit id #'.$message->getHitId(), ['message' => $message]);
+
+            throw new InvalidPayloadException('Invalid hit id #'.$message->getHitId(), (array) $message);
+        }
+
+        //  Lead IS mandatory field
+        if (null === $lead = $this->leadRepository->find($message->getLeadId())) {
+            $this->logger->error('Invalid lead id #'.$message->getLeadId(), ['message' => $message]);
+
+            throw new InvalidPayloadException('Invalid lead id', (array) $message);
+        }
+
+        return [
+            'hit'                    => $hit,
+            'page'                   => $pageObject,
+            'request'                => MessengerRequestFactory::fromArray($message->getRequest()),
+            'lead'                   => $lead,
+            'trackingNewlyGenerated' => $message->isNew(),
+            'activeRequest'          => false,
+            'hitDate'                => new \DateTime((new DateTimeHelper($message->getEventTime(), 'c'))->toLocalString()),
```

Comment:
```suggestion
            'hitDate'                => new \DateTimeImmutable((new DateTimeHelper($message->getEventTime(), 'c'))->toLocalString()),
```

See https://www.reddit.com/r/PHP/comments/w7v4pb/deprecating_the_mutable_datetime_class

Reply:
ha! good to know, thanks. I won't use it anymore.

Reply:
Why we use  local date ? Everwhere in app we working with UTC. This seems tiny

`(new DateTimeHelper($message->getEventTime(), 'c')->getUtcDateTime()`